### ISSUE TITLE
Some text strings seem to be no longer in use

### ISSUE THREAD

**Issue by Greg-21** [thexaero/Better-PVP-Translations#3]
UPDATED TO REFLECT THE STATE FROM BETTER PVP VERSION 21.16.0 AND 21.10.6.

With your Better PvP mod installed, I checked all the places I could think about, and in none of them I was able to find the text strings listed below.

Unused since a very old mod version (I've seen old screenshots with them on third-party websites that redistribute your mods):
* `gui.xaero_display_teams` (line 69) (this and the 4 below were superseded by toggles on the Entity Colors/Radar screen)
* `gui.xaero_display_players` (line 71)
* `gui.xaero_display_mobs` (line 72)
* `gui.xaero_display_items` (line 73)
* `gui.xaero_display_other` (line 74)
* `gui.xaero_entity_colours` (line 137) (superseded by `gui.xaero_entity_radar` (`Entity Radar`))

Unused since ver. 20.17.0:
* `gui.xaero_mp_cross_tp` (line 275)

Unused since ver. 20.20.0 (in this case, I'm just guessing after looking at the changelog):
* `gui.xaero_server_disabled` (line 36)

Unused since ver. 20.29.0:
* `gui.xaero_tiny` (line 97) (this and the 3 below were superseded by the more precise number-based selection of the minimap size)
* `gui.xaero_small` (line 98)
* `gui.xaero_medium` (line 99)
* `gui.xaero_large` (line 100)

Unused since ver. 21.3.0:
* `gui.xaero_slopes_default` (line 361) (superseded by `gui.xaero_slopes_default_3d` (`Default 3D`) and `gui.xaero_slopes_default_2d` (`Default 2D`))

Unused since ver. 21.15.0 (but 1.8 and 1.7 are stuck on an older mod version, so obviously they haven't received this update, and these text strings are still used there, see: #15):
* `gui.xaero_entity_radar` (line 224) (superseded by `gui.xaero_entity_radar_categories` (`Entity Category Settings`))
* `gui.xaero_box_cave_maps` (line 302) (superseded by `gui.xaero_box_cave_maps2`)
* `gui.xaero_entity_radar_type` (line 365) (superseded by `gui.xaero_entity_radar_category` (`Category:`))

Unused since ver. 21.16.0 (but again, the vast majority of these text strings already existed in the older mod version 1.8 and 1.7 are stuck on, so these text strings are still used there):
* `gui.xaero_waypoints_scale` (line 163)
* `gui.xaero_box_waypoints_scale` (line 396)
* `gui.xaero_box_distance` (line 305) (superseded by `gui.xaero_box_distance2`)
* `gui.xaero_auto_clamp_depth` (line 389)
* `gui.xaero_waypoints_clamp_depth` (line 390)
* `gui.xaero_box_waypoints_clamp_depth` (line 393)
* `gui.xaero_box_waypoints_distance_scale` (line 394)
* `gui.xaero_box_waypoints_name_scale` (line 395)

Lower probability that they are still in use (but I may be wrong):
* `gui.xaero_pvp_sprint` (line 101)
* `gui.xaero_fbo` (line 146)
* `gui.xaero_world` (line 177)
* `gui.xaero_waypoint_set` (line 178)
* `gui.xaero_display_hostile` (line 188)
* `gui.xaero_set_name` (line 272)

Higher probability that they are still in use (but I wasn't able to determine if and where):
* `gui.xaero_settings_search` (line 460)

In addition to all the above, there are a few more text strings that are no longer used in your Better PvP mod for Minecraft 1.12 and newer but are still used in versions like 1.8 and 1.7, which you still support. For more information, you can read #15.

**Comment by thexaero** [comment]
Thanks for the help!
I was already planning to go through the list and remove the unused lines. Some of the ones you posted are definitely used though. No need to make the list yourself, that would be a lot of work on your part. I'll probably make a script to go through the whole code and check every line.
I also want to make a clearer separation of the different Better PVP features, but that'll be a bit more work.

**Comment by Greg-21** [comment]
Which phrases are still in use? Could you please guide me to the places in the game where they are "hidden"? If I wasn't able to find them, probably other translators were unable to as well. Having a context helps a lot in the translation process.

> No need to make the list yourself, that would be a lot of work on your part. I'll probably make a script to go through the whole code and check every line.

If you mean a script that checks only translation files, that's what I already have. It moves lines around, removes the ones that don't match, and could also add missing lines if I implemented such feature (but I decided that it would be a stupid idea, since every change in the U.S. English translation would require a change in all other 24 (!) translation files as well), based on a specified pattern file. OK, end of this bragging (:

I planned to make a PR with such cleanup when I had some free time, but I was blocked by this issue we are talking about here, by the missing tooltip I reported you, and by the pending PRs with updated translations (the biggest blocker, as I definitely didn't want to cause merge conflicts that cannot be resolved by the GitHub web interface).

**Comment by Greg-21** [comment]
Which phrases are still in use? Could you please guide me to the places in the game where they are "hidden"? If I wasn't able to find them, probably other translators were unable to as well. Having a context helps a lot in the translation process.

> No need to make the list yourself, that would be a lot of work on your part. I'll probably make a script to go through the whole code and check every line.

If you mean a script that checks only translation files, that's what I already have. It moves lines around, removes the ones that don't match, and could also add missing lines if I implemented such feature (but I decided that it would be a stupid idea, since every change in the U.S. English translation would require a change in all other 24 (!) translation files as well), based on a specified pattern file. OK, end of this bragging (:

I planned to make a PR with such cleanup when I had some free time, but I was blocked by this issue we are talking about here, by the missing tooltip I reported to you, and by the pending PRs with updated translations (the biggest blocker, as I definitely didn't want to cause merge conflicts that cannot be resolved by the GitHub web interface).

**Comment by thexaero** [comment]
I help with the translations, so if they can't find context, then it's not an unsolvable issue.
No, that's not the script I described there. I'll make something like that to reorganize stuff though. And please don't do this yourself. I appreciate that you want to help, but it's definitely something I should do. As you have already found out, it's not very obvious where each line is, and you waste a lot of your time looking for them, whereas I instantly know what everything is and what is or isn't used.

**Comment by Greg-21** [comment]
Yeah, that's true, I can't see the source code. Every popular minimap mod for Minecraft is not open source, so I'm used to it. Also, I don't have enough free time anymore, and the task is quite extensive, so I wasn't even going to take it further this month. No worries.

**Comment by thexaero** [comment]
The list you put together is definitely helpful though! Thank you. I didn't realize there were so many unused lines. It adds up. I'll try to get to this sooner than I was probably previously going to.

**Comment by Greg-21** [comment]
Just a quick addition of things I haven't notice before. How? I have no idea :/

The following lines are already non-existent in the U.S. English lang file (from Better PvP for MC 1.12.2), but are still in the U.S. English JSON file (from Better PvP for MC 1.16.3):
* `"gui.xaero_set_name": "Set Name"`
* `"gui.xaero_visit_needed": "§eYou need to visit this dimension first to convert it to the new format!"`
* `"gui.xaero_mp_cross_tp": "MP Cross-Dimensional TP"`
* `"gui.xaero_cross_tp": "Cross-Dimensional TP"` (this one seems to be OK)
* `"gui.xaero_unreachable_dimension": "Unreachable destination dimension."`
* `"gui.xaero_mp_cross_disabled": "Cross-dimensional teleportation is disabled in multiplayer by default. Imperfections in the multi-world detection can easily kill your character if you teleport while in the wrong sub-server or world. You can enable it in the mod settings. Use at your own risk!"`

These are a bit different between the U.S. English lang file and the U.S. English JSON file (IMO the word "IPs" looks better when it is on the left):
* `gui.xaero_hide_world_names=Hide IPs/World Names`
* `"gui.xaero_hide_world_names": "Hide World Names/IPs"`

And the following one is in a different place (line 289 instead of 272):
* `gui.xaero_waypoint_name=Waypoint Name`

**Comment by thexaero** [comment]
Most of these don't exist in the 1.12.2 file because cross-dimensional teleportation wasn't a thing back then. But I do think the files should still be consistent now. 
_gui.xaero_set_name_ is the name of the text box, which is needed for the accessibility narrator. I'll actually add the ones that are still used.

**Comment by Greg-21** [comment]
I've just updated the original issue description at the top to reflect the current state from Better PvP version 21.15.0 (and 21.10.5).

**Comment by Greg-21** [comment]
Since all or almost all of the unused lines listed here have been removed from translation files, I'm closing this issue.
